WalletWasabi/Cache/IdempotencyRequestCache.cs

```diff
@@ -28,6 +27,18 @@ public IdempotencyRequestCache(IMemoryCache cache)
 	/// <remarks>Guarded by <see cref="ResponseCacheLock"/>.</remarks>
 	private IMemoryCache ResponseCache { get; }
 
+	/// <summary>
+	/// Try to get the value associated with the given key.
+	/// </summary>
+	public bool TryGetValue<TRequest, TResponse>(TRequest cacheKey, out TResponse? value)
```

Comment:
I think there is a problem with this function. It will return current state but at the next moment it can change. So if you base decisions to do the actual work, meanwhile you return and start getting the tx somebody else might finished it already. We did this in the past and ended up like that. The idea of `GetCachedResponseAsync` is perfect.